From 77705c80856f86662c3472b5b34da16d96da2ac4 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@opensourcefoundries.com>
Date: Thu, 19 Oct 2017 01:11:42 -0200
Subject: [PATCH] resize-helper.service: wait for udevadm settle before using
 udevadm info

Otherwise the ID_* variables will not be available in time for
resize-helper.

Signed-off-by: Ricardo Salveti <ricardo@opensourcefoundries.com>
---
 resize-helper.service | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/resize-helper.service b/resize-helper.service
index c1354a6..e3a0f8e 100644
--- a/resize-helper.service
+++ b/resize-helper.service
@@ -1,9 +1,11 @@
 [Unit]
 Description=Resize root filesystem to fit available disk space
-After=systemd-remount-fs.service
+Wants=systemd-udevd.service systemd-udev-trigger.service
+After=systemd-remount-fs.service systemd-udevd.service
 
 [Service]
 Type=oneshot
+ExecStartPre=-/bin/udevadm settle
 ExecStart=-/usr/sbin/resize-helper
 ExecStartPost=/bin/systemctl disable resize-helper.service
 
-- 
2.14.1

